helpers/lib_tasks_pytest.py

```diff
@@ -874,6 +874,121 @@ def run_coverage_report(  # type: ignore
+        - include_in_report is the glob pattern to include
+        - exclude_from_report is a comma‑separated glob pattern to omit, or ``None``
+    """
+    if target_dir == ".":
+        include_in_report: str = "*"
+        exclude_from_report: Optional[str] = None
+        if hserver.skip_submodules_test():
+            submodule_paths: List[str] = hgit.get_submodule_paths()
+            exclude_from_report = ",".join(
+                f"{path}/*" for path in submodule_paths
+            )
+    else:
+        include_in_report = f"*/{target_dir}/*"
+        exclude_from_report = None
+    return include_in_report, exclude_from_report
+
+
+def _run_coverage(
+    ctx,
+    suite: str,
+    target_dir: str,
+    generate_html_report: bool,
+) -> str:
+    """
+    Internal helper to run coverage for a given suite (fast/slow) and emit XML.
+
+    :param suite: one of "fast" or "slow"
+    :param target_dir: coverage target directory
+    :param generate_html_report: whether to produce HTML output
+    :return: filename of the suite‑specific coverage data file
+    """
+
+    if suite not in ("fast", "slow"):
+        raise ValueError(
+            f"Unsupported suite='{suite}'. Expected 'fast' or 'slow'."
+        )
+    test_cmd = f"invoke run_{suite}_tests --coverage -p {target_dir} -- --junitxml=reports/junit_{suite}.xml -o junit_family=legacy"
+    coverage_file = f".coverage_{suite}_tests"
+    hlitauti.run(ctx, test_cmd, use_system=False)
+    hsystem.system(f"mv .coverage {coverage_file}")
+    hdbg.dassert_file_exists(coverage_file)
+    include_in_report, exclude_from_report = _get_inclusion_settings(target_dir)
+    report_cmd: List[str] = [
+        "coverage erase",
+        f"coverage combine --keep {coverage_file}",
+    ]
+    report_stats_cmd: str = (
+        f"coverage report --include={include_in_report} --sort=Cover"
+    )
+    if exclude_from_report:
+        report_stats_cmd += f" --omit={exclude_from_report}"
+    report_cmd.append(report_stats_cmd)
+    if generate_html_report:
+        report_html_cmd: str = f"coverage html --include={include_in_report}"
+        if exclude_from_report:
+            report_html_cmd += f" --omit={exclude_from_report}"
+        report_cmd.append(report_html_cmd)
+    report_cmd.append("coverage xml -o coverage.xml")
+    full_report_cmd: str = " && ".join(report_cmd)
+    docker_cmd_: str = f"invoke docker_cmd --use-bash --cmd '{full_report_cmd}'"
+    hlitauti.run(ctx, docker_cmd_)
+    return coverage_file
+
+
+@task
+def run_fast_coverage(
+    ctx, target_dir: str, generate_html_report: bool = True
+) -> str:
+    """
+    Task wrapper to run *fast* test suite with coverage and emit reports.
+
+    :param ctx: Invoke context
+    :param target_dir: coverage target directory
+    :param generate_html_report: whether to produce HTML output
+    :return: filename of the fast‑suite coverage data file
+    """
+    return _run_coverage(ctx, "fast", target_dir, generate_html_report=False)
```

Comment:
Can we assign "fast" to variable and then pass it to the function? 

Same everywhere.

Reply:
It's a thin layer (the function body is one line and there is 20 lines around it)

But do you need to bind the task? If that's the reason, I'm ok with it